master/buildbot/changes/gitpoller.py

```diff
@@ -311,6 +312,16 @@ def _convert_nonzero_to_failure(res,
             "utility to handle the result of getProcessOutputAndValue"
             (stdout, stderr, code) = res
             if code != 0:
+                if code == 128:
+                    # 128 is the error code git returns when things are
+                    # wrong on its end
+                    error_message = ('RuntimeError: command %s %s in %s on '
+                                    'repourl %s failed with exit code %d: %s'
+                                    % (command, args, path, self.repourl,
+                                       code, stderr)).strip()
+                    sys.stderr.write(error_message)
+                    sys.stderr.flush()
+                    sys.exit(1)
```

Comment:
It is not the case. Buildbot is an agregation of a lot of services. An exception in one of them dont make the master shutdown. Only this service would be shutdown, and if there are other source of changes in the same master, or forced build, the master can still be available.

Here, as it looks like this comes from a misconfigured git server, I would suggest to log the error (with log.msg), and then retry the poll after a delay.